src/tree.cc

```diff
@@ -248,6 +260,80 @@ void Tree::Node::Swap(Tree::Node* destination) {
   std::swap(parent_, destination->parent_);
 }
 
+void Tree::Node::Resize(double delta) {
+  if (!parent_) {
+    return;
+  }
+
+  Tree::Node* sibling = GetRightSibling();
+  if (!sibling) {
+    sibling = GetLeftSibling();
+  }
+  if (!sibling || sibling->ratio_ < delta + min_ratio_ || this->ratio_ < -delta + min_ratio_) {
+    return;
+  }
+
+  this->ratio_ += delta;
+  sibling->ratio_ -= delta;
+
+  parent_->FitChildrenRatios();
+}
+
+// Resize this node to specific ratio and distribute the remaining space to siblings evenly.
+void Tree::Node::ResizeToRatio(double ratio) {
+  if (!parent_) {
```

Comment:
Maybe we could return early if the given ratio is >= 1?
Since there wouldn't be any remaining space to distribute.

```suggestion
  if (!parent_ || ratio >= 1.) {
```